### PR TITLE
Move /connect/account-abstraction to app router

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces a new `Account Abstraction` feature in the dashboard, enhancing user experience with smart wallets. It includes new components, API key management, and alerts to facilitate integration and usage of account abstraction in applications.

### Detailed summary
- Removed `account-abstraction.tsx` file.
- Added `getAASupportedAPIKeys` function to fetch API keys.
- Created `AccountAbstractionAPIKeysMenu` for API key selection.
- Implemented `PageHeader` with documentation link.
- Introduced `Layout` for consistent page structure.
- Added `OpChainAlert` for gas credits notifications.
- Updated `SmartWallets` to accept `defaultTab` prop.
- Created `Page` component to handle API keys and redirects.
- Developed a new page for handling client-specific actions with `clientId`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/page.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <AccountAbstractionPage
+      defaultClientId={clientId}
+      apiKeys={apiKeys}
+      dashboardAccount={account}
+    />
+  );
+}
+
+export const dynamic = "force-dynamic";
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/page.tsx:34]: needed?

Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/page.tsx]: removed

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/[clientId]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    tab?: string;
+  };
+}) {
+  const { clientId } = props.params;
+  const dashboardAccount = await getAccount();
+
+  if (!dashboardAccount) {
+    redirect(
+      `/login?next=${encodeURIComponent(
+        `/dashboard/connect/account-abstraction${clientId}`,
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/[clientId]/page.tsx:26]: missing`/` before `clientId`?

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/[clientId]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  );
+
+  return (
+    <div className="flex flex-col gap-10">
+      <div className="flex flex-col content-start justify-between gap-4 lg:flex-row">
+        <PageHeader />
+
+        <div>
+          <AccountAbstractionAPIKeysMenu
+            apiKeys={apiKeys}
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/[clientId]/page.tsx:53]: only pass the pieces u need?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/[clientId]/page.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/account-abstraction/[clientId]/page.tsx]: updated

**Review by jnsdls — APPROVED** [review]

**Pull request merged**